Source/Core/Core/Slippi/SlippiMatchmaking.cpp

```diff
@@ -289,19 +288,31 @@ void SlippiMatchmaking::startMatchmaking()
 
   ERROR_LOG(SLIPPI_ONLINE, "[Matchmaking] Trying to find match...");
 
-  // if (!m_user->IsLoggedIn())
-  // {
-  //   ERROR_LOG(SLIPPI_ONLINE, "[Matchmaking] Must be logged in to queue");
-  //   m_state = ProcessState::ERROR_ENCOUNTERED;
-  //   m_errorMsg = "Must be logged in to queue. Go back to menu";
-  //   return;
-  // }
+  /*if (!m_user->IsLoggedIn())
+  {
+      ERROR_LOG(SLIPPI_ONLINE, "[Matchmaking] Must be logged in to queue");
+      m_state = ProcessState::ERROR_ENCOUNTERED;
+      m_errorMsg = "Must be logged in to queue. Go back to menu";
+      return;
+  }*/
+
+  // The following code attempts to fetch the LAN IP such that when remote IPs match, the
+  // LAN IP can be tried in order to establish a connection in the case where the players
+  // don't have NAT loopback which allows that type of connection.
+  // Right now though, the logic would replace the WAN IP with the LAN IP and if the LAN
+  // IP connection didn't work but WAN would have, the players can no longer connect.
+  // Two things need to happen to improtve this logic:
+  // 1. The connection must be changed to try both the LAN and WAN IPs in the case of
+  //    matching WAN IPs
+  // 2. The process for fetching LAN IP must be improved. For me, the current method
+  //    would always fetch my VirtualBox IP, which is not correct. I also think perhaps
+  //    it didn't work on Linux/Mac but I haven't tested it.
+  // I left this logic on for now under the assumption that it will help more people than
+  // it will hurt
+  char lan_addr[30] = "";
 
-  // Compute LAN IP, in case 2 people are connecting from one IP we can send them each other's local
-  // IP instead of public. Experimental to allow people from behind one router to connect.
   char host[256];
-  char lan_addr[30];
-  char* ip;
+  char* ip = static_cast<char*>(malloc(16));
   struct hostent* host_entry;
   int hostname;
   hostname = gethostname(host, sizeof(host));  // find the host name
```

Comment:
this whole section of socket programming looks generally out of date. most of the functions called are deprecated and don't support ipv6

good sample code to lift from: https://stackoverflow.com/questions/52727565/client-in-c-use-gethostbyname-or-getaddrinfo

Reply:
might be worth rewriting this all eventually with boost asio or something to make this less esoteric and get away from the c-isms